### ISSUE TITLE
Property type Decimal instead of Double

### ISSUE THREAD

**Issue by benrudhart** [yonaskolb/SwagGen#184]
My yaml defines the property of a model with type `number`.
This generates the Swift type `Double`.

I'm looking for a way to automatically generate a Swift `Decimal`. 
I thought something like this should work, but format is `ignored` in this case (generates Swift `Double`)
```
 Foo:
      properties:
        bar:
          type: number
          format: decimal
```

**Comment by lemoinem** [comment]
We have the same issue.
At the very least, it would be nice to have an option to choose whether to use float/double or decimal type for non-integer numbers.

**Comment by benrudhart** [comment]
any updates on this one? @lemoinem how do you handle this?

**Comment by lemoinem** [comment]
@benrudhart Unfortunately, our solution was to use something other than SwagGen for our code generation needs. Sorry.

**Comment by benrudhart** [comment]
mh I see. @lemoinem which generator are you using now?

**Comment by nickaroot** [comment]
We really need it

**Comment by yonaskolb** [comment]
Support for this has been added in https://github.com/yonaskolb/SwagGen/pull/259 and the just released 4.4.0
